File: app/src/main/java/vergecurrency/vergewallet/service/repository/GeocodingService.java
package vergecurrency.vergewallet.service.repository;

import org.json.JSONException;
import org.json.JSONObject;

public class GeocodingService {

	//TODO : Query should be done here
	//TODO : Make it a static class
	public GeocodingService() {
	}

	public String readCoordinates(String rawData) {

		String lat = "";
		String lon = "";


		if (rawData != null && !rawData.equals("")) {
			JSONObject reader;
			try {
				reader = new JSONObject(rawData);
				lat = reader.getString("latitude");
				lon = reader.getString("longitude");
				return String.format("%s;%s", lat, lon);
			} catch (JSONException e) {
				e.printStackTrace();
				return "error";
			}
		} else return "error";
	}
}

File: app/src/main/kotlin/vergecurrency/vergewallet/service/repository/GeocodingService.kt
package vergecurrency.vergewallet.service.repository

import cz.msebera.android.httpclient.client.methods.HttpGet
import org.json.JSONException
import org.json.JSONObject
import vergecurrency.vergewallet.Constants
import vergecurrency.vergewallet.service.model.network.layers.NetworkGateway

object GeocodingService {

    fun readCoordinates(ipAddress: String): String {

        var rawData: String = NetworkGateway().doRequest(HttpGet(String.format(Constants.IP_DATA_ENDPOINT, ipAddress)))

        if (rawData != "") {
            val reader: JSONObject
            try {
                reader = JSONObject(rawData)
                val lat = reader.getString("latitude")
                val lon = reader.getString("longitude")
                return String.format("%s;%s", lat, lon)
            } catch (e: JSONException) {
                e.printStackTrace()
                return "error"
            }

        } else
            return "error"
    }
}
